### ISSUE TITLE
used before being assigned - ts(2454)

### ISSUE THREAD

**Issue by y-nk** [microsoft/TypeScript#45715]
# Bug Report

### 🔎 Search Terms

ts2454 await

### 🕗 Version & Regression Information

- I was unable to test this on prior versions because i only use latest version

### ⏯ Playground Link

[Playground Link](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAtgQwJZhgXhgiBPMwYAUAlOgHwwDeAUDLTADYCmsAZmAFwwBiArnlEnBUadBAHdksMIzEwACgCcQcJBEYAeAG4gkAE1IEFjCCHqbGZSiLp026QiTTkKAX2s2YRk2cbF3LomEbNj8XIA)

### 💻 Code

```ts
const main = async () => {
    let fn: Function

    await new Promise<void>(resolve => {
        fn = () => {}
        resolve()
    })

    fn()
}
```

### 🙁 Actual behavior

error ts(2454) is detected

### 🙂 Expected behavior

error ts(2454) shouldn't be detected since it should be detected that the promise is awaited for.

**Comment by MartinJohns** [comment]
Duplicate of #9998. I wonder when we'll cross the 9998 duplicates mark.
